File: c/questao1.c
#include <stdio.h>
#include <string.h>

int findFirstOccurrence(char* str, char c, int startIdx) {
    int len = strlen(str);

    for (int i = startIdx; i < len; i++) {
        if (str[i] == c) {
            return i;
        }
    }

    return -1; // Caractere não encontrado
}

int main() {
    char str[100];
    char c;
    int startIdx;

    printf("Digite a string: ");
    scanf("%s", str);

    printf("Digite o caractere a ser buscado: ");
    scanf(" %c", &c);

    printf("Digite a posição inicial: ");
    scanf("%d", &startIdx);

    int result = findFirstOccurrence(str, c, startIdx);

    if (result != -1) {
        printf("O caractere '%c' foi encontrado pela primeira vez na posição %d da string.\n", c, result);
    } else {
        printf("O caractere '%c' não foi encontrado na string a partir da posição %d.\n", c, startIdx);
    }

    return 0;
}

File: python/questao1.py
def find_first_occurrence(string, char, start_idx):
    length = len(string)

    for i in range(start_idx, length):
        if string[i] == char:
            return i

    return -1  # Caractere não encontrado

def main():
    string = input("Digite a string: ")
    char = input("Digite o caractere a ser buscado: ")
    start_idx = int(input("Digite a posição inicial: "))

    result = find_first_occurrence(string, char, start_idx)

    if result != -1:
        print(f"O caractere '{char}' foi encontrado pela primeira vez na posição {result} da string.")
    else:
        print(f"O caractere '{char}' não foi encontrado na string a partir da posição {start_idx}.")

if __name__ == "__main__":
    main()
